huxley/core/tests/models/test_models.py

```diff
@@ -174,10 +178,29 @@ def test_update_assignments(self):
 
         Assignment.update_assignments(updates)
         new_assignments = [a[1:] for a in Assignment.objects.all().values_list()]
-
+        delegates = Delegate.objects.all()
         updates = [(cm.id, ct.id, s.id, rej) for cm, ct, s, rej in updates]
         self.assertEquals(set(updates), set(new_assignments))
+        self.assertEquals(len(delegates), 0)
 
+    def test_update_assignment(self):
+        '''Tests that when an assignment changes schools, its rejected
+           field is set to False and any delegates assigned to it are
+           no longer assigned to it.'''
+        s1 = TestSchools.new_school(name='S1')
+        s2 = TestSchools.new_school(name='S2')
+        a = TestAssignments.new_assignment(school=s1, rejected=True)
+        d1 = TestDelegates.new_delegate(school=s1, assignment=a)
+        d2 = TestDelegates.new_delegate(school=s1, assignment=a)
+
+        a.school = s2
+        a.save()
+
+        delegates = Delegate.objects.all()
```

Comment:
You can also refer back to the delegates from assignments, using the `related_name` param of the ForeignKey field on delegates:
- https://docs.djangoproject.com/en/1.8/topics/db/queries/#following-relationships-backward
- https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.ForeignKey.related_name

So you can do something with the `a.delegates` QuerySet. I believe QuerySet has an `exists` or `count` operation, but you can double-check the docs.